src/Databases/IDatabase.cpp

```diff
@@ -18,7 +21,13 @@ StoragePtr IDatabase::getTable(const String & name, ContextPtr context) const
 {
     if (auto storage = tryGetTable(name, context))
         return storage;
-    throw Exception(ErrorCodes::UNKNOWN_TABLE, "Table {}.{} doesn't exist", backQuoteIfNeed(getDatabaseName()), backQuoteIfNeed(name));
+    TableNameHints hints(const_cast<IDatabase *>(this)->shared_from_this(), context);
```

Comment:
it's better to add `using ConstDatabasePtr = std::shared_ptr<const IDatabase>;` in IDatabase.h and use it as a parameter for TableNameHints constructor